app/controllers/registrations_controller.rb

```diff
@@ -33,50 +33,57 @@ def create
     else
       build_resource(devise_parameter_sanitizer.sanitize(:sign_up))
       clean_up_passwords(resource)
-      set_flash_message :alert, :captcha_error
+      set_flash_message(:alert, :captcha_error)
       render :new
     end
   end
 
   def destroy
     password_correct = resource.valid_password?(deletion_params[:password])
-    if !password_correct
-      set_flash_message :alert, :password_incorrect
-      respond_with_navigational(resource) {
+    unless password_correct
+      set_flash_message(:alert, :password_incorrect)
+      respond_with_navigational(resource) do
         redirect_to after_sign_up_path_for(resource_name)
-      }
+      end
       return
     end
     success = resource.archive_and_destroy(deletion_params[:archive_name])
-    if !success
-      set_flash_message :alert, :not_destroyed
-      respond_with_navigational(resource) {
+    unless success
+      set_flash_message(:alert, :not_destroyed)
+      respond_with_navigational(resource) do
         redirect_to after_sign_up_path_for(resource_name)
-      }
+      end
       return
     end
     Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
-    set_flash_message :notice, :destroyed
-    yield resource if block_given?
-    respond_with_navigational(resource) {
+    set_flash_message(:notice, :destroyed)
+    yield(resource) if block_given?
+    respond_with_navigational(resource) do
       redirect_to after_sign_out_path_for(resource_name)
-    }
+    end
   end
 
-  def after_sign_up_path_for(resource)
+  def after_sign_up_path_for(_resource)
     edit_profile_path
   end
 
   private
 
     def check_registration_limit
-      if User.where("users.confirmed_at is NULL and users.created_at > '#{(DateTime.now() - (ENV['MAMPF_REGISTRATION_TIMEFRAME'] || 15).to_i.minutes)}'").count > (ENV['MAMPF_MAX_REGISTRATION_PER_TIMEFRAME'] || 40).to_i
-        self.resource = resource_class.new devise_parameter_sanitizer.sanitize(:sign_up)
-        resource.validate # Look for any other validation errors besides reCAPTCHA
-        set_flash_message :alert, :too_many_registrations
-        set_minimum_password_length
-        respond_with_navigational(resource) { render :new }
-      end
+      timeframe = (ENV["MAMPF_REGISTRATION_TIMEFRAME"] || 15).to_i.minutes
+      threshold_date = DateTime.now - timeframe
+      new_users = User.where(
+        "users.confirmed_at is NULL and users.created_at > '#{threshold_date}'"
+      ).count
+
+      max_users = (ENV["MAMPF_MAX_REGISTRATION_PER_TIMEFRAME"] || 40).to_i
+      return if new_users > max_users
+
```

Comment:
Are you sure you want `>` this way around? The way I read the old code, the statements following this line were executed if `new_users > max_users` which would imply that in the guard clause we have to `return if new_users < max_users` (or `<=` if you want to exactly preserve the old behaviour).

Reply:
Oh, I should've looked more carefully here. You are absolutely right. Just fixed it with commit cb2fa144ef.